Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -2412,9 +2412,11 @@ void RenderBlockFlow::addFloatsToNewParent(RenderBlockFlow& toBlockFlow) const
         toBlockFlow.createFloatingObjects();
 
     for (auto& floatingObject : m_floatingObjects->set()) {
-        if (toBlockFlow.containsFloat(floatingObject->renderer()))
-            continue;
-        toBlockFlow.m_floatingObjects->add(floatingObject->cloneForNewParent());
+        if (floatingObject->hasValidRenderer()) {
+            auto& renderer = floatingObject->renderer();
+            if (!toBlockFlow.containsFloat(renderer))
+                toBlockFlow.m_floatingObjects->add(floatingObject->cloneForNewParent());
+        }
```

Comment:
Not sure that just adding a null check is a great idea without understanding why it’s null. This kind of speculative fix usually means we don’t really know whether we are fixing the bug or not. If it’s reproducible under Address Sanitizer it seems we have an opportunity to debug it and make a reduced test case and fix it for real.

That having been said, if it is a practical step to add a null check there’s no need to refactor the surrounding code, and the refactor doesn’t seem to be an improvement, adding a local variable we only use once and changing to avoid “early `continue`”, which doesn’t seem necessary. We can write this which is just adding two lines of code and leaving the rest as is:
```suggestion
        if (!floatingObject->hasValidRenderer()))
            return;
        if (toBlockFlow.containsFloat(floatingObject->renderer()))
            continue;
        toBlockFlow.m_floatingObjects->add(floatingObject->cloneForNewParent());
```